### PR TITLE
Working draft [fast&dirty]

### PR DESCRIPTION
**Pull request by zhenik**
I will do rebase, once get approved on this PR.

There are a bunch of TODOs.

Screenshot 2020-08-26 at 13 24 00

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -3,10 +3,11 @@ Vagrant.configure("2") do |config|
     config.vm.box_version = ">= 0.3, < 0.4"
     config.vm.provider "virtualbox" do |vb|
         vb.linked_clone = true
-        vb.memory = 2048
+        vb.memory = 4096
+        vb.cpus = 4
```
Comment (fredrikhgrelland) [inline Vagrantfile:7]: Consider less here. Not all computers can affort 4 cores. 3 might be more appropriate

File: format.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+terraform fmt -recursive
```
Comment (fredrikhgrelland) [inline format.sh:2]: Should be upstreamed to template.

File: Makefile [hunk 3]
```diff
@@ -28,13 +31,20 @@ example: custom_ca
...
-destroy-box:
+destroy:
 	vagrant destroy -f
+	rm terraform.tfstate
+	rm terraform.tfstate.backup
+	rm example/terraform.tfstate
+status:
+	vagrant global-status
+format:
+	./format.sh
```
Comment (fredrikhgrelland) [inline Makefile:47]: Upstream to template

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,44 @@
+locals {
+  datacenters     = join(",", var.nomad_datacenters)
+  hive_env_vars   = join("\n",
+    concat([
+      "YO=man",
```
Comment (fredrikhgrelland) [inline main.tf:5]: Should not be necessary.

File: test_example/example/nomad_acl_test.tf [hunk 5]
```diff
@@ -1,7 +1,7 @@
 data "vault_generic_secret" "nomad_secret_id" {
   # Set count of this data source to 1 if ACLs are enabled in Nomad, and 0 if not
   count = var.nomad_acl ? 1 : 0
-  path = "nomad/creds/write"
+  path  = "nomad/creds/write"
```
Comment (fredrikhgrelland) [inline test_example/example/nomad_acl_test.tf:4]: Change upstream as well

File: test_example/main.tf [hunk 6]
```diff
@@ -1,4 +1,4 @@
 resource "nomad_job" "countdash" {
   jobspec = file("${path.module}/conf/nomad/countdash.hcl")
-  detach = false
+  detach  = false
```
Comment (fredrikhgrelland) [inline test_example/main.tf:3]: Change upstream as well

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    service_name  = string,
+    port          = number,
+    database_name = string,
+    username      = string,
+    password      = string
+  })
+  description = "Postgres service name and port"
+  default = ({
+    service_name  = ""
+    port          = 0
```
Comment (fredrikhgrelland) [inline variables.tf:57]: Why not a default port?

Comment (fredrikhgrelland) [reply variables.tf:57]: And in general. Defaults that are erronous should not be defaults. These should not be dafaults if you actually need to implement something to use the module.

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    port         = number,
+    access_key   = string,
+    secret_key   = string,
+  })
+  description = "Minio service object"
+  default = ({
+    service_name = ""
+    port         = 0
+    access_key   = "",
+    secret_key   = "",
```
Comment (fredrikhgrelland) [inline variables.tf:77]: same ^

File: Makefile [hunk 9]
```diff
@@ -28,13 +31,20 @@ example: custom_ca
 ifdef CI # CI is set in Github Actions
 	cd test_example; SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant up --provision
 else
-	cp -f docker/conf/certificates/*.crt test_example/docker/conf/certificates
+	if [ -f "docker/conf/certificates/*.crt" ]; then cp -f docker/conf/certificates/*.crt test_example/docker/conf/certificates; fi
```
[the remark this answers is not in the document]
Comment (zhenik) [reply Makefile:34]: yeah, I know. Just wonder about sync with the upstream branch. Do we agree about a strategy on how to sync with the template ?

File: Makefile [hunk 10]
```diff
@@ -28,13 +31,20 @@ example: custom_ca
...
 
 # clean commands
-destroy-box:
+destroy:
 	vagrant destroy -f
+	rm terraform.tfstate
+	rm terraform.tfstate.backup
+	rm example/terraform.tfstate
+status:
+	vagrant global-status
```
[the remark this answers is not in the document]
Comment (zhenik) [reply Makefile:45]: I would say yes, but it is still a draft. Let us leave as it is for now. I have feelings that more changes are coming in that context

File: Vagrantfile (hunk 1 above)
Comment (zhenik) [reply Vagrantfile:7]: I think I can start with 2 core

File: format.sh (hunk 2 above)
Comment (zhenik) [reply format.sh:2]: lets postponed it for now

File: variables.tf (hunk 7 above)
Comment (zhenik) [reply variables.tf:57]: Yeah. I also think it should be required without default values.

File: Makefile (hunk 9 above)
[the remark this answers is not in the document]
Comment (fredrikhgrelland) [reply Makefile:34]: No, not yet i guess.

**Comment by zhenik** [comment]
Check https://github.com/fredrikhgrelland/terraform-nomad-hive/pull/1/checks?check_run_id=1031439299#step:8:379 is failing due to `minio` module. I have to remove [`provider "nomad"` from main.tf there](https://github.com/fredrikhgrelland/terraform-nomad-minio/blob/master/main.tf#L1-L3)

**Comment by zhenik** [comment]
Blocker https://github.com/fredrikhgrelland/terraform-nomad-minio/pull/8

**Pull request merged**